### DISCUSSION TITLE
Remote Jupyter Kernel Command Execution

### DISCUSSION THREAD

**Discussion by EliotKhachi** [jupyter/jupyter_server#1408] [Q&A]
Hi everyone,

I'm very new to anything beyond just using jupyter notebooks, but I've found myself needing to dive into the innards for an application I'm working on.

I'm trying to send commands from my application to a remote jupyter kernel and get the kernel print statements back.

I couldn't find what I was looking for in [the jupyter server REST API docs](https://jupyter-server.readthedocs.io/en/latest/developers/rest-api.html). Do I need to look into [Server Extensions](https://jupyter-server.readthedocs.io/en/latest/developers/extensions.html)?

**AM I ON THE RIGHT TRACK?**

Thanks,
Eliot

**Comment by EliotKhachi** [comment]
What about Jupyter Kernel Gateway? From my digging it seems like its better suited for remote interactions.

**Reply by gsnoopy** [reply]
The Jupyter Kernel Gateway is a tool that allows remote execution of Jupyter Notebook kernels via HTTP APIs. You can interact with notebooks using HTTP requests, which is useful for integrating notebooks with your app

I found [this article](https://towardsdatascience.com/expose-endpoints-using-jupyter-kernel-gateway-e55951b0f5ad) teaching how to use the Jupyter Kernel Gateway. It works as well. You can create endpoints with Flask if you're using Python, or with Express in JavaScript, and make requests with Requests in Python or Axios in JavaScript.

Basically, you:

1. Create the notebook you want to access.
2. Execute it.
3. Define the endpoints to access the data from your notebook.
4. Make HTTP requests to access the data.

Remember to install Jupyter Kernel Gateway using: pip install jupyter_kernel_gateway
And start the Jupyter Kernel Gateway server using: jupyter kernelgateway --port 8888
